pkg/lokifrontend/frontend/v2/frontend.go

```diff
@@ -204,6 +204,7 @@ enqueueAgain:
 
 	select {
 	case <-ctx.Done():
+		level.Warn(f.log).Log("msg", "request enqueued, but not cancelable", "queryID", freq.queryID)
```

Comment:
I requested this because I have doubt this is practical and if it is we're enqueuing without being able to cancel which can lead to the same bug as before.